### DISCUSSION TITLE
history-graph could be more meaningful with fixed scale option

### DISCUSSION THREAD

**Discussion by pooh22** [home-assistant/frontend#18258] [Dashboard]
The current `history-graph` card doesn't seem to have the option to define a fixed scale, eg min=0 max=2, to define the bottom and top values in the graph. Even better would be min="value at left time" and max="value at left time"+2

Specific use-cases would be where the graph plots something that you consume (power, water, heat) and you expect certain normal usage values. Unless there's really unusual behaviour, the graph plots a more or less straight line from bottomleft to topright because the ranges are derived from the data. 
This doesn't give me more information than that the meter-reading is working (good to know!), but what would help me understand the usage, would be to set limits for how much is to be expected normally, then I can see how fast or slow the usage is over the day. And if it's a high usage or low usage day, for example.

So for a history graph with a constantly growing datasource, it would make sense to define a value range per timeperiod and use that to define the vertical range of the graph displayed. For other types of graph a fixed min/max value may be more suitable, but I can't think of an example right now.

**Comment by ildar170975** [comment]
Are you talking about a possibility to define a minimal Y-axis boundary?
![изображение](https://github.com/home-assistant/frontend/assets/71872483/25af563e-a37a-43dd-83d7-2514f90b3843)
Imho this is mainly important in part of a "lower boundary".
Agree, having a possibility to define the "lower boundary" in `history-graph` would be useful.

As a workaround:
1. Use a custom card like "mini-graph-card" which allows to define boundaries.
2. (cumbersome) Use card-mod - the 2nd card above was made by it ([explained here](https://community.home-assistant.io/t/card-mod-add-css-styles-to-any-lovelace-card/120744/5147)).

**Reply by pooh22** [reply]
I mean something similar, but e.g when you are plotting the value of the gas, or in my case heat, meter, it continually grows, so it's at 32 and grows by 0.1 per day (or something). The graph would look like a straight horizontal line if you just put minimal value of 0.

To continue with the heat example. In winter the usage per day is like 1GJ per day, but in summer only 0.010 GJ, but in the current graph, both look almost the same. When the graph for 1 day would have a maximum vertical range of 1 or 1.5 GJ, you can spot immediately if it's summer or winter from the graph. Better still, you could spot differences in usage patterns better, because it is more to scale of the expected.

**Reply by ildar170975** [reply]
Currently the `history-graph` is "self-adapted" to a data range within a selected time period.
Assume, on summer the data range is "100 units", on winter it is "50 units" - but graphs look similar (similar curves, except they represent different data ranges).
So, only a manual defining of a "lower boundary" may help you. You may define it to a value which will cause both "seasons" to be displayed with visible changes (not a flatline) + these changes will be seen differently for "summer" & "winter".

If so - you still may use workarounds described above (even with an `input_number` to define a lower boundary when using a "card-mod way"). Until the corresponding option is added to `history-graph`.

**Reply by pooh22** [reply]
in the mini-card-graph there is a parameter: `min_bound_range_secondary`, which looks like it might fit my issue, though I haven't tried it yet and the implemented in version 0.x.x appears to imply it will be implemented at some time in the future.

The option to plot a second graph with the minimum boundary doesn't solve my problem (0 is so far out of range the graph will be a straight horizontal line at the top). Perhaps if I manually adjust the value every few days, but I'm not going to do that, I expect ;-)

I hope something like a `bound range` option can be added to the history-graph card, I think it would be a useful option to have for values that represent a consumption like energy/heat/water which can have a predetermined expected maximum consumption rate. It could even be used to help usage reduction targets by setting it lower than usual and try to stay below the value.

**Reply by ildar170975** [reply]
https://github.com/home-assistant/frontend/pull/19297

**Reply by Jojo-A** [reply]
I also think that an option to set upper and lower y-boundaries manually should be implemented into the standard history-graph card.
Of course auto-scaling is a good approach as a "first shot". But for meaningful interpretation it is essential to scale the desired signal into the "real world". While when you have only auto-scaling you might see only a zoomed version of the noise of the signal.
Example room temperature:
the actual value ranges from 19°C to 21°C. dT = 2°C. Relative to the interesting range of 10°C to 40C° this is **~6%** "visual variation".
But the graph scales it in the range of 17°C to 23°C, which results in **33%** "visual variation"! This is pointless.
A fundamental key competence when gathering data is to interpret them correctly. And this interpretation requires a meaningful reference system.
